store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -378,6 +394,62 @@ protected ItemActionResult handleTrashOperation(OperationContext octxt, Element
         return trashResults;
     }
 
+    /**
+     * @param octxt
+     * @param mbox
+     * @param local
+     * @param type
+     * @param context
+     * @throws ServiceException
+     * @throws MessagingException Method to delete the files on sharer side when
+     *                            owner file is deleted also notify user that file
+     *                            access has been revoked
+     */
+    protected void handleTrashOperationForSharedFile(OperationContext octxt, Mailbox mbox, List<Integer> local,
+            MailItem.Type type, Map<String, Object> context) throws ServiceException, MessagingException {
+        ZimbraSoapContext zsc = getZimbraSoapContext(context);
+        Account authAccount = getAuthenticatedAccount(zsc);
+        Account account = getRequestedAccount(zsc);
+        MailItem[] items = mbox.getItemById(octxt, local, MailItem.Type.UNKNOWN);
+        SendShareNotification sendNotification = new SendShareNotification();
+        for (MailItem item : items) {
+            if (item instanceof Document && item.getACL() != null) {
+                for (Grant grant : item.getACL().getGrants()) {
+                    // form shareinfodata object to send file revoke notification email to each grantee
+                    ShareInfoData sr = new ShareInfoData();
+                    sr.setOwnerAcctId(item.getAccountId());
+                    sr.setOwnerAcctEmail(mbox.getAccount().getMail());
+                    sr.setOwnerAcctDisplayName(mbox.getAccount().getDisplayName() == null ? mbox.getAccount().getMail()
+                            : mbox.getAccount().getDisplayName());
+                    sr.setItemId(item.getId());
+                    sr.setItemUuid(item.getUuid());
+                    sr.setPath(item.getPath());
+                    sr.setFolderDefaultView(Type.DOCUMENT);
+                    sr.setGranteeType(ACL.GRANTEE_USER);
+                    // get grantee email Id to details to which notification email will be sent
+                    Account granteeAccount = Provisioning.getInstance().getAccount(grant.getGranteeId());
+                    sr.setGranteeId(grant.getGranteeId());
+                    sr.setGranteeName(granteeAccount.getName());
+                    MimeMessage mm = null;
```

Comment:
don't see need for null initialization here as its directly getting initialized on next line, can use `sendNotification.generateShareNotification`

Reply:
updated accordingly